openbas-front/src/components/common/queryable/uri/useUriState.tsx

```diff
@@ -1,44 +1,58 @@
 import * as qs from 'qs';
-import { useLocation, useNavigate } from 'react-router-dom';
-import { useEffect, useState } from 'react';
+import { useSearchParams } from 'react-router-dom';
+import { useEffect, useRef, useState } from 'react';
 import * as R from 'ramda';
 import { z } from 'zod';
 import { UriHelpers } from './UriHelpers';
 import type { SearchPaginationInput } from '../../../../utils/api-types';
 import { buildSearchPagination, SearchPaginationInputSchema } from '../QueryableUtils';
 
+export const retrieveFromUri = (localStorageKey: string, searchParams: URLSearchParams): SearchPaginationInput | null => {
+  const encodedParams = searchParams.get('query') || '';
+  const params = atob(encodedParams);
+  const paramsJson = qs.parse(params, { allowEmptyArrays: true }) as unknown as SearchPaginationInput & { key: string };
+  if (!R.isEmpty(paramsJson) && paramsJson.key === localStorageKey) {
+    try {
+      const parse = SearchPaginationInputSchema.parse(paramsJson);
+      return buildSearchPagination(parse);
+    } catch (err) {
+      if (err instanceof z.ZodError) {
+        // eslint-disable-next-line no-console
+        console.log(`Validation error: the uri has not a valid format ${err.issues}`);
+        return null;
+      }
+    }
+  }
+  return null;
+};
```

Comment:
we make retriveFromUri a function taht we can call directly to avoid to initialize the hook and doing a double request